pkg/operator/apiserver/controller/workload/workload.go

```diff
@@ -102,139 +98,41 @@ func NewController(name, operatorNamespace, targetNamespace, targetOperandVersio
-		utilruntime.HandleError(fmt.Errorf("caches did not sync"))
-		return
-	}
-
-	// doesn't matter what workers say, only start one.
-	go wait.Until(c.runWorker, time.Second, ctx.Done())
-
-	<-ctx.Done()
-}
-
-func (c *Controller) runWorker() {
-	for c.processNextWorkItem() {
-	}
-}
-
-func (c *Controller) processNextWorkItem() bool {
-	dsKey, quit := c.queue.Get()
-	if quit {
-		return false
-	}
-	defer c.queue.Done(dsKey)
-
-	err := c.sync()
-	if err == nil {
-		c.queue.Forget(dsKey)
-		return true
-	}
-
-	utilruntime.HandleError(fmt.Errorf("%v failed with : %v", dsKey, err))
-	c.queue.AddRateLimited(dsKey)
-
-	return true
-}
-
-// AddInformer queues the given informer to check spec, status and managed resources
-func (c *Controller) AddInformer(informer cache.SharedIndexInformer) *Controller {
-	informer.AddEventHandler(c.eventHandler())
-	c.preRunCachesSynced = append(c.preRunCachesSynced, informer.HasSynced)
-	return c
-}
-
-// AddNamespaceInformer queues the given ns informer for the targetNamespace
-func (c *Controller) AddNamespaceInformer(informer cache.SharedIndexInformer) *Controller {
-	interestingNamespaces := sets.NewString(c.targetNamespace)
-	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc: func(obj interface{}) {
-			ns, ok := obj.(*corev1.Namespace)
-			if !ok {
-				c.queue.Add(workQueueKey)
-			}
-			if interestingNamespaces.Has(ns.Name) {
-				c.queue.Add(workQueueKey)
-			}
-		},
-		UpdateFunc: func(old, new interface{}) {
-			ns, ok := old.(*corev1.Namespace)
-			if !ok {
-				c.queue.Add(workQueueKey)
-			}
-			if interestingNamespaces.Has(ns.Name) {
-				c.queue.Add(workQueueKey)
-			}
-		},
-		DeleteFunc: func(obj interface{}) {
-			ns, ok := obj.(*corev1.Namespace)
-			if !ok {
-				tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
-				if !ok {
-					utilruntime.HandleError(fmt.Errorf("couldn't get object from tombstone %#v", obj))
-					return
-				}
-				ns, ok = tombstone.Obj.(*corev1.Namespace)
-				if !ok {
-					utilruntime.HandleError(fmt.Errorf("tombstone contained object that is not a Namespace %#v", obj))
-					return
-				}
-			}
-			if interestingNamespaces.Has(ns.Name) {
-				c.queue.Add(workQueueKey)
-			}
-		},
-	})
-	c.preRunCachesSynced = append(c.preRunCachesSynced, informer.HasSynced)
-
-	return c
-}
-
-func (c *Controller) eventHandler() cache.ResourceEventHandler {
-	return cache.ResourceEventHandlerFuncs{
-		AddFunc:    func(obj interface{}) { c.queue.Add(workQueueKey) },
-		UpdateFunc: func(old, new interface{}) { c.queue.Add(workQueueKey) },
-		DeleteFunc: func(obj interface{}) { c.queue.Add(workQueueKey) },
-	}
-}
-
 // shouldSync checks ManagementState to determine if we can run this operator, probably set by a cluster administrator.
-func (c *Controller) shouldSync(operatorSpec *operatorv1.OperatorSpec) (bool, error) {
+func (c *Controller) shouldSync(controllerContext factory.SyncContext, operatorSpec *operatorv1.OperatorSpec) (bool, error) {
```

Comment:
we could pass only the event recorder, wdyt?

Reply:
yeah, nothing else is needed from the controller context :+1:

I've also noticed I missed a `context.TODO()` in this func, I'll fix that with the new revision